client/my-sites/theme/main.jsx

```diff
@@ -71,48 +67,17 @@ const ThemeSheet = React.createClass( {
 	},
 
 	getInitialState() {
-		return { selectedAction: null };
+		return {
+			showPreview: false,
+		};
 	},
 
 	componentDidMount() {
 		window.scroll( 0, 0 );
 	},
 
-	hideSiteSelectorModal() {
-		this.setState( { selectedAction: null } );
-	},
-
-	isActive() {
-		const { id, currentTheme } = this.props;
-		return currentTheme && currentTheme.id === id;
-	},
-
-	onPrimaryClick() {
-		if ( ! this.props.isLoggedIn ) {
-			this.props.signup( this.props );
-		} else if ( this.isActive() ) {
-			this.props.customize( this.props, this.props.selectedSite );
-		} else if ( this.props.price ) {
-			this.selectSiteAndDispatch( 'purchase' );
-		} else {
-			this.selectSiteAndDispatch( 'activate' );
-		}
-	},
-
-	onPreviewButtonClick() {
-		if ( ! this.props.isLoggedIn ) {
-			this.props.signup( this.props );
-		} else {
-			this.selectSiteAndDispatch( 'customize' );
-		}
-	},
-
-	selectSiteAndDispatch( action ) {
-		if ( this.props.selectedSite ) {
-			this.props[ action ]( this.props, this.props.selectedSite, 'showcase-sheet' );
-		} else {
-			this.setState( { selectedAction: action } );
-		}
+	getTheme() {
+		return pick( this.props, [ 'id', 'name', 'author', 'screenshot', 'price', 'description', 'descriptionLong', 'supportDocumentation', 'download', 'taxonomies', 'stylesheet', 'active', 'purchased' ] );
```

Comment:
> Looking at it again, my concern is the dependencies (eg: `customize()` from `state/themes/actions`, `isPremium()`, `selectedTheme` prop from `ThemesSiteSelectorModal`) which accept a theme object. IMO we should probably either standardize the theme object for these things or have them accept simpler props (eg: just theme slug).

While we won't be able to standardize all function signatures (e.g. have all functions from that group accept exactly one theme and one site argument), I've started a PR to turn our theme helpers into selectors (which I believe they ultimately are), and which will clean up things a bit as it will enable us to bind to `site` early, i.e. at `connect` stage.

Tangentially, I'd like then to change all theme selectors to accept theme IDs (instead of entire objects), as that will make them more versatile (e.g for edge cases like [this](https://github.com/Automattic/wp-calypso/blob/8a20ad81021ebb87bef3b3b4aac341b209c6caca/client/me/purchases/product-link/index.jsx#L41)). I think that style is starting to become accepted practice (cf site related selectors) -- if really needed, a selector can always get the corresponding object itself.